### DISCUSSION TITLE
Displaying a Custom Image from cv.Mat or PIL Image

### DISCUSSION THREAD

**Discussion by schmiddey** [flet-dev/flet#142] [Q&A]
Is it possible to show an image only existing in ram? like an opencv image or a PIL image? but not saving it before.

I want do to show a camera stream, where i only get single images. (ROS)

**Comment by FeodorFitsner** [comment]
Would displaying an image from base64 work for your case?

**Reply by schmiddey** [reply]
Yea that should work.
So i need to put the base64 buffer as the Image src?

Thank you :) sorry that is not my area of expertise

**Comment by FeodorFitsner** [comment]
Yes, now it's possible with `Image.src_base64` property. Additionally, you can configure `assets` directory and save images there and then re-load them in the app by adding something like `?ts={timestamp}` to the image URL.

**Reply by schmiddey** [reply]
Thank you so much :D it works fine

**Reply by FeodorFitsner** [reply]
Nice! Would be cool if you could share a simple example with us :)

**Comment by ultranity** [comment]
As the encoding process will introduce extra cost, it will still be very nice if flet could pass the original mem buffter to the dart Control as it will be converted back to a Uint8List. Especially if we want to implement real time video stream, which requires encoding & docoding.

Since Dart, Python, and Go supports C-API model, it should be possible to go over the gap. And there exists some libs like Darpy.
> Darpy is a library that allows dart to call out to python code by using the Python C-API embedding and dart:ffi.

I am still reading the src code of Flet and I wonder if this issue is related to the C/S arch and IPC protocol design.

**Reply by FeodorFitsner** [reply]
Hm, it's [Dartpy](https://github.com/TimWhiting/dartpy) and it looks like it could be a way to a standalone packaging. I see it was a one-time effort/experiment 12 months ago and the library hasn't been evolving since then. If we'd able to call Python directly from Flutter then I'm sure we could elimintate the need of Go server in between - I have a couple of ideas how to do that. Thanks for the link. As I understand to use the same approach on the phones we need to build ARM64 version of Python. I'll be trying to deal with that this year.
